src/app_service/service/token/async_tasks.nim

```diff
@@ -9,25 +9,29 @@ import ../../../backend/backend as backend
 
 type
   GetTokenDetailsTaskArg = ref object of QObjectTaskArg
-    chainId: int
+    chainIds: seq[int]
     address: string
 
 
 const getTokenDetailsTask*: Task = proc(argEncoded: string) {.gcsafe, nimcall.} =
   let arg = decode[GetTokenDetailsTaskArg](argEncoded)
-  try:
-    let response = backend.discoverToken(arg.chainId, arg.address).result
+  for chainId in arg.chainIds:
+    try:
+      let response = backend.discoverToken(chainId, arg.address).result
+      
+      let output = %* {
+        "address": arg.address,
+        "name": response{"name"}.getStr,
+        "symbol": response{"symbol"}.getStr,
+        "decimals": response{"decimals"}.getInt
+      }
+      arg.finish(output)
+      return
+    except Exception as e:
+      continue
     
-    let output = %* {
-      "address": arg.address,
-      "name": response{"name"}.getStr,
-      "symbol": response{"symbol"}.getStr,
-      "decimals": response{"decimals"}.getInt
-    }
-    arg.finish(output)
-  except Exception as e:
-    let output = %* {
-      "address": arg.address,
-      "error": "Is this an ERC-20 or ERC-721 contract?",
-    }
-    arg.finish(output)
+  let output = %* {
```

Comment:
@alaibe this line won't be reached ever if `arg.chainIds.len` is greater than 0 and `backend.discoverToken` is successfully called (because of `return` in the line `29`).

Reply:
This will be reach if no chain have this token, they all will raise an exception which will cause the call on continue rather than return